packages/clay-tooltip/src/TooltipProvider.tsx

```diff
@@ -139,121 +157,204 @@ const TooltipProvider: React.FunctionComponent<{
 
-			dispatch({type: 'hide'});
-			clearTimeout(timeoutIdRef.current);
+				titleTag.remove();
+			}
 		}
-	}, []);
+	};
+
+	const restoreTitle = () => {
+		const element = titleNodeRef.current;
+
+		if (element) {
+			const title = element.getAttribute('data-restore-title');
 
-	const handleEsc = useCallback((event: KeyboardEvent) => {
-		if (event.key === Keys.Esc) {
-			event.stopImmediatePropagation();
-			handleHide(event);
+			if (title) {
+				if (element.tagName === 'svg') {
+					const titleTag = document.createElement('title');
+
+					titleTag.innerHTML = title;
+
+					element.appendChild(titleTag);
+				} else {
+					element.setAttribute('title', title);
+				}
+
+				element.removeAttribute('data-restore-title');
+			}
+
+			titleNodeRef.current = null;
 		}
-	}, []);
+	};
+
+	const handleShow = ({target}: any) => {
+		targetRef.current = target;
 
-	const handleShow = useCallback(({target}: any) => {
-		const hasTitle = target && target.hasAttribute('title');
+		const hasTitle =
+			target &&
+			(target.hasAttribute('title') || target.hasAttribute('data-title'));
 
 		const titleNode = hasTitle
 			? target
-			: closestAncestor(target, '[title]');
+			: closestAncestor(target, '[title], [data-title]');
 
-		const title = titleNode && titleNode.getAttribute('title');
+		if (titleNode) {
+			const title =
+				titleNode.getAttribute('title') ||
+				titleNode.getAttribute('data-title');
 
-		if (title) {
-			titleNodeRef.current = titleNode;
-			targetRef.current = target;
-
-			document.addEventListener('keyup', handleEsc, true);
-			target.addEventListener('click', handleHide);
-			titleNode.setAttribute('data-title', title);
-			titleNode.removeAttribute('title');
+			saveTitle(titleNode);
 
 			const customDelay = titleNode.getAttribute('data-tooltip-delay');
 			const newAlign = titleNode.getAttribute('data-tooltip-align');
+			const setAsHTML = titleNode.getAttribute('data-title-set-as-html');
 
 			timeoutIdRef.current = setTimeout(
 				() => {
 					dispatch({
 						align: newAlign || align,
 						message: title,
+						setAsHTML,
 						type: 'show',
 					});
 				},
 				customDelay ? Number(customDelay) : delay
 			);
 		}
+	};
+
+	const handleHide = () => {
+		dispatch({type: 'hide'});
+
+		clearTimeout(timeoutIdRef.current);
+
+		restoreTitle();
+
+		targetRef.current = null;
+	};
+
+	React.useEffect(() => {
+		const handleEsc = (event: KeyboardEvent) => {
+			if (show && event.key === Keys.Esc) {
+				event.stopImmediatePropagation();
+
+				handleHide();
+			}
+		};
+
+		document.addEventListener('keyup', handleEsc, true);
+
+		return () => document.removeEventListener('keyup', handleEsc, true);
+	}, [show]);
+
+	React.useEffect(() => {
+		let disposeShowEvents: Array<any> = [];
+		let disposeHideEvents: Array<any> = [];
+
+		if (scope) {
+			disposeShowEvents = TRIGGER_SHOW_EVENTS.map((eventName) => {
+				return delegate(document.body, eventName, scope, handleShow);
```

Comment:
does it make sense to use the delegate here? just thinking about avoiding the use of `closest` twice, we will use it in `handleShow` and we have the `closest` to `delegateTarget`, I also don’t know if it makes sense because we won’t use this information, this is usually a little expensive. What do you think?

Reply:
Hmm, I'm not totally sure what you mean by `we won’t use this information`. The delegates here are the primary means to using the event listener as a singleton and scoping to a specific selector. These delegates should only be used in the context of a singleton and when we aren't wrapping the react children with onMouseOver and onMouseOut.

`.closest` is used in `delegate()` because we have to know if we are within the given selector. If the selector is `.my-cool-app`, every element inside of that would emit an event for our events (`mouseout` & `mouseover`). We also call `.closest` again in our event callback to check if any parents in the tree have a `title` or `data-title` attribute. This is necessary because of something like `<button title="test"><svg /></button>`, if we don't check up the tree, hovering the svg wouldn't show the tooltip.

Does that help clarify a bit? It is a little expensive, but I'm not sure if there is a good way to get around it...

Reply:
Oh yeah, I think I got the `delegate` implementation completely wrong 🤦‍♂️, it makes sense to keep that.